internal/brokers/broker.go

```diff
@@ -191,11 +187,12 @@ func (b Broker) cancelIsAuthorized(ctx context.Context, sessionID string) {
 //	        }
 //	    }
 //	}
-func generateValidators(supportedUILayouts []map[string]string) (map[string]map[string]fieldValidator, error) {
+func generateValidators(ctx context.Context, supportedUILayouts []map[string]string) map[string]map[string]fieldValidator {
 	validators := make(map[string]map[string]fieldValidator)
 	for _, layout := range supportedUILayouts {
 		if _, exists := layout["type"]; !exists {
-			return nil, fmt.Errorf("Supported UI layouts are invalid")
+			log.Errorf(ctx, "layout %v provided with missing type, it will be ignored", layout)
```

Comment:
hum, we are missing some context in the log, which session ID this has been provided? I think then we have to bubble the error up and handling logging in the caller. However, as said in the previous one, it means that theorically, we should not trust the fieldValidator returned value.

So… maybe pass the sessionID just for the sake of logging? Not ideal, but I don’t have a better idea atm :/

Reply:
We could attach the sessionID to the context we pass for this function, wdyt?

Reply:
That’s one way I was thinking about too. Unfortunately, it means that we are going to again have to pass the context everywhere we want to log in. As we are not as deep as in adsys, I think passing the value is the most straightforward way for now. Let’s revisit if we get this case in more places if you don’t mind.